samples/minimum_criteria.rst

```diff
@@ -0,0 +1,96 @@
+.. _definition_and_minimum_criteria:
+
+Sample Definition and Minimum Criteria
+######################################
+
+Definition
+==========
+
+A sample is a basic Zephyr application that provides a brief overview of a feature, subsystem,
+or module. For example, Kernel services, drivers, protocols, etc.
+Samples are limited in scope and should focus only on capturing the nontrivial and introductory
+aspects of the functionality in question.
```

Comment:
@microbuilder I think for providing an "overview of one or more features, subsystems" there is a proposal for the "Demo" category https://github.com/zephyrproject-rtos/zephyr/issues/47872. It may be confusing to distinguish between the two. Thoughts?

Reply:
I wasn't aware of this distinction, thanks.

Reply:
IMO a demo is just a more detailed and feature sample, and since we do not have this category, we can promote samples to demos if they qualify as something we want to use to demonstrate multiple features at the same time. I agree with suggested change by @microbuilder

Reply:
@microbuilder @nashif Updated the definition to reflect the suggested change.